sdk/highlight-go/middleware/echo/middleware.go

```diff
@@ -0,0 +1,48 @@
+package echo
+
+import (
+	"context"
+	"strings"
+
+	"github.com/highlight/highlight/sdk/highlight-go"
+	"github.com/highlight/highlight/sdk/highlight-go/middleware"
+	"github.com/labstack/echo/v4"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+// Middleware is a echo compatible middleware
+// use as follows:
+//
+// import highlightecho "github.com/highlight/highlight/sdk/highlight-go/middleware/echo"
+// ...
+// r.Use(highlightecho.Middleware())
+
+func Middleware() echo.MiddlewareFunc {
+	middleware.CheckStatus()
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) error {
+			ctx := c.Request().Context()
+			highlightReqDetails := c.Request().Header.Get("X-Highlight-Request")
+			ids := strings.Split(string(highlightReqDetails), "/")
+			if len(ids) >= 2 {
+				ctx = context.WithValue(ctx, highlight.ContextKeys.SessionSecureID, ids[0])
+				ctx = context.WithValue(ctx, highlight.ContextKeys.RequestID, ids[1])
+			}
+
+			span, hCtx := highlight.StartTrace(ctx, "highlight/echo")
+			defer highlight.EndTrace(span)
+
+			c.SetRequest(c.Request().Clone(hCtx))
```

Comment:
does this inject `hCtx` as the echo handler's context?

Reply:
Yes, but there may be a cleaner way to do this. I'll check.